File: mongo/rhino/read/mongo_read.js
#! /usr/bin/java org.mozilla.javascript.tools.shell.Main
// --------------------------------------------------------------
//	read/mongo_read.js
//
//					Sep/02/2013
// --------------------------------------------------------------
importPackage(com.mongodb);
// --------------------------------------------------------------
print ("*** 開始 ***");

var mm = new Mongo ("localhost",27017);

var db = mm.getDB ("city_db" );

var col_name = "saitama";
 
var coll = db.getCollection (col_name);

mongo_display_proc (coll);

print ("*** 終了 ***");

// --------------------------------------------------------------
function mongo_display_proc (coll)
{
	print (coll.getCount());

	var cur = coll.find();

	while(cur.hasNext())
		{
		var obj = cur.next();

		var out_str = obj.get ("key") + "\t";
		out_str += obj.get ("name") + "\t";
		out_str += obj.get ("population") + "\t";
		out_str += obj.get ("date_mod");
		print (out_str);
		}
}

// --------------------------------------------------------------

File: mongo/jruby/read/mongo_read.rb
#! /usr/bin/jruby
# -*- coding: utf-8 -*-
#
#	mongo_read.rb
#					Sep/02/2013
#
# --------------------------------------------------------
include Java
import com.mongodb.DB
import com.mongodb.Mongo
import com.mongodb.DBCollection
import com.mongodb.DBCursor

# --------------------------------------------------------
require 'rubygems'
require 'json'
# --------------------------------------------------------
puts	"*** 開始 ***"

mm = Mongo.new("localhost",27017)
db = mm.getDB("city_db" )

col_name = "saitama"

coll = db.getCollection(col_name)


puts coll.getCount()

cursor = coll.find()

while cursor.hasNext() do
	data = cursor.next()
	unit_aa=JSON.parse(data.to_s)
	print(unit_aa['key'],"\t")
	print(unit_aa['name'],"\t")
	print(unit_aa['population'],"\t")
	print(unit_aa['date_mod'],"\n")
	end

puts "*** 終了 ***"
# --------------------------------------------------------
